src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -240,6 +264,54 @@ ZoneSet::MoveWindowIntoZoneByIndex(HWND window, HWND windowZone, int index) noex
     }
 }
 
+IFACEMETHODIMP_(void)
+ZoneSet::MoveWindowIntoZoneByIndexSet(HWND window, HWND windowZone, const std::vector<int>& indexSet) noexcept
+{
+    if (m_zones.empty())
+    {
+        return;
+    }
+
+    while (auto zoneDrop = ZoneFromWindow(window))
+    {
+        zoneDrop->RemoveWindowFromZone(window, !IsZoomed(window));
+    }
+
+    if (indexSet.size() == 1)
+    {
+        MoveWindowIntoZoneByIndex(window, windowZone, indexSet[0]);
+        return;
+    }
+
+    RECT size;
+    bool sizeEmpty = true;
+
+    for (int index : indexSet)
+    {
+        if (index < static_cast<int>(m_zones.size()))
+        {
+            RECT newSize = m_zones.at(index)->ComputeActualZoneRect(window, windowZone);
```

Comment:
```suggestion
            RECT newSize = m_zones[index]->ComputeActualZoneRect(window, windowZone);
```
nit: would be a bit faster, since you've already checked the bounds